Documentation/contributing/release/organization.rst

```diff
@@ -31,5 +23,69 @@ for the next stable release. New patch releases for an existing stable version
 New patch releases for stable branches are made periodically to provide
 security and bug fixes, based upon community demand and bugfix severity.
 Potential fixes for an upcoming release are first merged into the ``main``
-branch, then backported to the relevant stable branches using the following
-criteria.
+branch, then backported to the relevant stable branches according to the
+:ref:`backport_criteria`.
+
+The following sections describe in more detail the general guidelines that the
+release management team follows for Cilium. The team may diverge from this
+process at their discretion.
+
+Feature Releases
+~~~~~~~~~~~~~~~~
+
+There are several key dates during the feature development cycle of Cilium
+which are important for developers:
+
+* Prerelease days: The Cilium release management team aims to publish a
+  snapshot of the latest changes in the ``main`` branch on the first weekday of
+  each month. This provides developers a target delivery date to incrementally
+  ship functionality, and allows community members to get early access to
+  upcoming features to test and provide feedback. Prereleases may not be
+  published when a release candidate or final stable release is being
+  published.
+
+* Feature freeze: Around six weeks prior to a target feature release, the
+  ``main`` branch is frozen for new feature contributions. The goal of the
+  freeze is to focus community attention on stabilizing and hardening the
+  upcoming release by prioritizing bugfixes, documentation improvements, and
+  tests. In general, all new functionality that the community intends to
+  distribute as part of the upcoming release must land into the ``main`` branch
+  prior to this date. Any bugfixes, docs changes, or testing improvements can
+  continue to be merged as usual following this date.
+
+* Release candidates: Immediately following the feature freeze, the release
+  management team aims to publish a release candidate. These candidates should
```

Comment:
```suggestion
  management team aims to publish release candidates. These candidates should
```

Reply:
:thinking: I guess the part about "Immediately" would contradict this suggestion, but given the title is `Release candidates`, it makes sense to adjust this wording a bit. I think I'll take your suggestion here and also drop the `Immediately` word.